### ISSUE TITLE
There is no content in some Task1_xxx.txt of result_dota

### ISSUE THREAD

**Issue by seekFire** [yijingru/BBAVectors-Oriented-Object-Detection#76]
After running the following command, I got 15 Task1_xxx.txt in both result_dota and merge_dota, but some of them has no content...

`python3 main.py --data_dir /media/Dota4BBAVectors/val --resume model_50.pth --conf_thresh 0.1 --dataset dota --phase eval`

I use val set of DOTA which has been split into 17651 patches(600*600) as input data. Do you think what's the reason of this phenomenon?

**Comment by yijingru** [comment]
I think maybe the image names do not follow
"""
P0000__0.5__0___0
P0000__0.5__0___1000
P0000__0.5__0___1500
"""
but you may debug to see the details.

**Comment by seekFire** [comment]
Maybe it is not the reason, the val image names after cropping by **ImgSplit_multi_process.py** are shown as follows:
```
P0003__0.5__0___0
P0003__1__0___0
P0003__1__0___423
P0003__1__500___0
P0003__1__500___423
P0003__1__547___0
P0003__1__547___423
```

The configuration in **ImgSplit_multi_process.py** is as follows:

```
split = splitbase(r'/media/val',
                       r'/media/Dota4BBAVectors/val',
                      gap=100,
                      subsize=600,
                      num_process=8
                      )
    split.splitdata(1)
    split.splitdata(0.5)
```
Thus the 458 original images were split into 17651 patches. I found that when I run the command mentioned above again and again, the Task1_xxx.txt without content has some difference with previous time...

**Comment by seekFire** [comment]
@yijingru I think I've known the reason. I had canceled the comment of '**BatchNorm2d**' in ctxbox_net.py, so the model became unstable. Now, everything goes well. Thank you!

**Comment by Lg955** [comment]
> @yijingru I think I've known the reason. I had canceled the comment of '**BatchNorm2d**' in ctxbox_net.py, so the model became unstable. Now, everything goes well. Thank you!

I remember uncomment the `BatchNorm2d in ctxbox_net.py` will  help stable training in #20 , but in this issue, "canceled the comment of '**BatchNorm2d**' in ctxbox_net.py,  the model became unstable",   so should the BN be used or not finally?

**Comment by seekFire** [comment]
@Lg955 
Well, because I used the pre-trained model which not contains the **BatchNorm2d** layer. If you want to use **BatchNorm2d**,
you may need to train a new model from scratch.

**Comment by Lg955** [comment]
> @Lg955
> Well, because I used the pre-trained model which not contains the **BatchNorm2d** layer. If you want to use **BatchNorm2d**,
> you may need to train a new model from scratch.

Oh, I see, thank U!

**Comment by Lg955** [comment]
@seekFire 
I have other problems about dataset:
1.After using `ImgSplit.py` , we will get trainsplit、valsplit and testsplit（the test data does not have labels in DOTA-V1.0), but as the author’s example:

              data_dir/ 
                images/*.png
                labelTxt/*.txt
                trainval.txt
                test.txt
the imgs in trainsplit、valsplit and testsplit are merged into `images/*.png`?

2.the `trainval.txt` means that the **train data** and **val data** in DOTA-V1.0 are both used to train(not valid after each epoch)?
Because in `train.py`, the eval not be used after each epoch in train mode:
https://github.com/yijingru/BBAVectors-Oriented-Object-Detection/blob/7efd410f5f4ded94aca986aeb0bd1292235d1314/train.py#L144

**Comment by seekFire** [comment]
@Lg955 
The 1st question:  Yes, you need to use `ImgSplit.py` for the three dataset separately then merge images & labels which come from train set & val set into directory **trainval**.
The 2nd question: Yes, data in directory **trainval** is only used for training.
